backend/src/server/server.go

```diff
@@ -25,8 +26,7 @@ import (
 // @contact.email	oduneye.d@northeastern.edu and ladley.g@northeastern.edu
 // @host 127.0.0.1:8080
 // @BasePath /
-// @schemes http https
-func Init(db *gorm.DB, settings config.Settings) *fiber.App {
+func Init(db *gorm.DB, pinecone search.PineconeClientInterface, settings config.Settings) *fiber.App {
```

Comment:
```suggestion
func Init(db *gorm.DB, settings config.Settings) *fiber.App {
```
does `pinecone search.PineconeClientInterface` make more sense to be stored by composition in settings? to avoid incrementing the len of the list of arguments each time we need a new dependency